src/test/kotlin/cacophony/utils/FileInputTest.kt

```diff
@@ -0,0 +1,311 @@
+package cacophony.utils
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import java.io.File
+
+class FileInputTest {
+    private fun createFileWithContent(content: String): String {
+        val file = File.createTempFile("test_file", ".cac")
+        file.deleteOnExit()
+        file.writeText(content)
+        return file.path
+    }
+
+    @Test
+    fun next() {
```

Comment:
I don't like the naming of these test functions (next(), peek(), etc.), short name seems like test helper function, maybe "next() simple" or something like that would be better.